instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/auto/instrumentation/opentelemetryapi/trace/UnshadedTracerProvider.java

```diff
@@ -42,4 +53,20 @@ public Tracer get(final String instrumentationName, final String instrumentation
             .get(instrumentationName, instrumentationVersion),
         contextStore);
   }
+
+  // this is called by OpenTelemetrySdk, which expects to get back a real TracerProviderSdk
+  @Override
+  public Object unobfuscate() {
+    if (!messageAlreadyLogged.getAndSet(true)) {
+      String message =
+          "direct usage of the OpenTelemetry SDK is not supported when running agent"
```

Comment:
Adding the concrete operation can provide good context

Direct usage of the OpenTelemetry SDK by calling OpenTelemetrySdk.getTracrProvider() is not...

Reply:
Also maybe a link to how to configure the agent (presumably what they were trying to do).